app/deliver_grant_funding/routes/reports.py

```diff
@@ -732,11 +732,24 @@ def edit_question(grant_id: UUID, question_id: UUID) -> ResponseReturnValue:
 @has_grant_role(RoleEnum.ADMIN)
 @auto_commit_after_request
 def manage_guidance(grant_id: UUID, question_id: UUID) -> ResponseReturnValue:
-    question = get_question_by_id(question_id=question_id)
+    question = get_component_by_id(component_id=question_id)
     form = AddGuidanceForm(obj=question)
 
     if form.validate_on_submit():
-        update_question(question, guidance_heading=form.guidance_heading.data, guidance_body=form.guidance_body.data)
+        # todo: both of these are equivalent as this is a property of the underlying component
+        #       should there be an update that handles either
+        if question.is_group:
```

Comment:
Quite frustrating that Python doesn't give us a nice way of using this :( I suppose you found `TypeGuard`/`TypeIs` and, like me just now, also discovered that they don't work as properties/methods on a class 🥲